go/grpcweb/wrapper.go

```diff
@@ -22,6 +24,10 @@ var (
 	}
 )
 
+const grpcContentType = "application/grpc"
+const grpcWebContentType = "application/grpc-web"
+const grpcWebTextContentType = "application/grpc-web-text"
```

Comment:
I've added a link to https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md#protocol-differences-vs-grpc-over-http2

These constants are actually used to search and replace the `Content-Type` header from the underlying gRPC server, so I think we do not want to include the `+whatever` suffixes. In theory: we could use use gRPC-Web with thrift or flatbuffers or whatever. I *think* this implementation will work, because it will just search/replace `application/grpc+flatbuffers` into `application/grpc-web-text+flatbuffers` and do the correct base64-encoding.

Reply:
I mention it because it seems we've hardcoded `+proto` in the client code.

Reply:
Oh good point, I had not noticed that. I am *far* from knowledgeable enough about gRPC, gRPC-web, or this implementation to have a good answer here. I just know enough to have implemented a Python server, and hacked support into `grpcwebproxy` that seems to work :)

Now that I look at this briefly, I *think* this mismatch is fine: The Go server part is a proxy around the "official" gRPC server, which attempts to be at least somewhat protocol agnostic. The grpc-web client in this repository is opinionated, and implements a type of `export interface Client<TRequest extends ProtobufMessage, TResponse extends ProtobufMessage>`, so it will only ever be used with the `+proto` variant.

For what it is worth, this is [documented as being the default if missing](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md#protocol-differences-vs-grpc-over-http2). The "official" upstream client seems to omit the `+proto` string, in either the "native" node implementation (which seems to default to `application/grpc`), or the grpc-web Javascript (which defaults to `application/grpc-web-text`)

Reply:
Indeed, I agree with you, so let's leave it as it is :).